acceptance-tests/tests/src/test/java/org/hyperledger/besu/tests/acceptance/privacy/multitenancy/MultiTenancyAcceptanceTest.java

```diff
@@ -0,0 +1,313 @@
+    multiTenancyCluster.start(node);
+    token = node.execute(permissioningTransactions.createSuccessfulLogin("user", "pegasys"));
+    node.useAuthenticationTokenInHeaderForJsonRpc(token);
+  }
+
+  @Test
+  public void privGetPrivacyPrecompileAddressShouldReturnExpectedAddress() {
+    node.verify(priv.privGetPrivacyPrecompileAddressSuccess(DEFAULT_PRIVACY));
+  }
+
+  @Test
+  public void privGetPrivateTransactionSuccessShouldReturnExpectedPrivateTransaction()
+      throws JsonProcessingException {
+    final PrivateTransaction validSignedPrivateTransaction =
+        getValidSignedPrivateTransaction(senderAddress);
+
+    receiveEnclaveStub(getRLPOutput(validSignedPrivateTransaction));
+    retrievePrivacyGroupEnclaveStub();
+    sendEnclaveStub("testKey");
+
+    final Hash transactionHash =
+        node.execute(
+            privacyTransactions.sendRawTransaction(
+                getRLPOutput(validSignedPrivateTransaction).encoded().toHexString()));
+    node.verify(priv.getTransactionReceiptSuccess(transactionHash));
+    node.verify(
+        priv.privGetPrivateTransactionSuccess(transactionHash, validSignedPrivateTransaction));
+  }
+
+  @Test
+  public void privCreatePrivacyGroupSuccessShouldReturnNewId() throws JsonProcessingException {
+    createPrivacyGroupEnclaveStub();
+
+    node.verify(
+        priv.privCreatePrivacyGroupSuccess(
+            List.of(KEY1, KEY2, KEY3), "GroupName", "Group description.", PRIVACY_GROUP_ID));
+  }
+
+  @Test
+  public void privDeletePrivacyGroupSuccessShouldReturnId() throws JsonProcessingException {
+    retrievePrivacyGroupEnclaveStub();
+    deletePrivacyGroupEnclaveStub();
+
+    node.verify(priv.privDeletePrivacyGroupSuccess(PRIVACY_GROUP_ID));
+  }
+
+  @Test
+  public void privFindPrivacyGroupSuccessShouldReturnExpectedGroupMembership()
+      throws JsonProcessingException {
+    final List<PrivacyGroup> groupMembership =
+        List.of(
+            testPrivacyGroup(emptyList(), PrivacyGroup.Type.BESU),
+            testPrivacyGroup(emptyList(), PrivacyGroup.Type.BESU),
+            testPrivacyGroup(emptyList(), PrivacyGroup.Type.BESU));
+
+    findPrivacyGroupEnclaveStub(groupMembership);
+
+    node.verify(priv.privFindPrivacyGroupSuccess(groupMembership.size(), ENCLAVE_KEY));
+  }
+
+  @Test
+  public void eeaSendRawTransactionSuccessShouldReturnPrivateTransactionHash()
+      throws JsonProcessingException {
+    final PrivateTransaction validSignedPrivateTransaction =
+        getValidSignedPrivateTransaction(senderAddress);
+
+    retrievePrivacyGroupEnclaveStub();
+    sendEnclaveStub("testKey");
+    receiveEnclaveStub(getRLPOutput(validSignedPrivateTransaction));
+
+    CompletableFuture<Hash> completableFuture = new CompletableFuture<>();
+
+    node.verify(
+        priv.eeaSendRawTransactionSuccess(
+            getRLPOutput(validSignedPrivateTransaction).encoded().toHexString(),
+            completableFuture));
+
+    completableFuture.whenComplete(
```

Comment:
added another comment about this, but would be good if we can avoid passing a future into the condition